### DISCUSSION TITLE
Tab title shows shell alias instead of actual expanded command

### DISCUSSION THREAD

**Discussion by edrozenberg** [ghostty-org/ghostty#3798] [Feature Requests, Ideas] [answered]
Ghostty Version 1.0.0
Build 8469
Commit 4b4d4062
(MacOS)

When using a shell alias for ex. `ss1 = ssh root@server1`, Ghostty shows `ss1` (the alias typed by the user) in the tab/window title.

Suggestion: Ghostty should show the actual command that the alias expanded to as the title, in this case `ssh root@server1`.

(Terminal.app shows the expanded command)

**Comment by jcollie** [comment, marked as answer]
Ghostty shows the title that the shell tells Ghostty to display. You'll have to dig into your shell config to see why it displays something different.

**Reply by edrozenberg** [reply]
That's fair, I was going off what Terminal.app does, which is that it automatically expands any alias to the alias' value in the tab title. But just looking at what Terminal.app does isn't a guarantee of correctness, it could just be an Apple thing or Apple "magic" behind the scenes that isn't available to other apps necessarily.

**Reply by jcollie** [reply]
Ghostty wouldn't even have access to your aliases to expand them.

**Reply by edrozenberg** [reply]
Was curious about other terms so looked briefly at iTerm2 (Mac), which for my alias example shows "ssh" as the tab title, so it is at least able to pick up on the command being executed, and possibly show parameters as well with additional config (but unconfirmed so far). Alactritty showed only "Alacritty" as the title ootb, probably with config it supports some aspects of custom tab titles but don't know to what extent yet.

I'll keep following Ghostty, if it later adds separate window vs tab titles, shell history/search/scrollbars will definitely be a great drop-in replacement for Terminal.app, which is still what I'm most used to. Ghostty already adds a lot of things for sure, at least one of which I use and which Terminal.app cannot do and never could do - the ability to use the Insert key in mc, vim etc., there's just no way to map Insert in Terminal.app, without reprogramming my keyboard or using a function key instead of the actual Ins key.
